builds.js

```diff
@@ -113,6 +113,57 @@ class Builds {
     return this.warehouse;
   }
 
+  /**
+   * PUT a (pre-)build set of files. The files have to be present on disk.
+   *
+   * @param {string[]} files Full pathnames of files to read from disk.
+   * @param {Object} params Parameters that specify pkg, env, version and/or locale.
+   * @param {Function} fn Completion callback.
+   * @public
+   */
+  async put(files, params, fn) {
+    const { env, locale, pkg, version } = this._readParams(params);
+
+    if (!pkg || !env || !version || !files || !Array.isArray(files)) {
+      return fn(new Error('Invalid parameters supplied'));
+    }
+
+    // Read file content and prepare a npm-like body with file attachments.
+    const name = params.pkg;
+    await this.warehouse.files.getAttachments(files).then(attachments => {
```

Comment:
if we are using `.then` to make this callback based to be consistent with other routes still, we shouldnt need this await or this to be an async function.